modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/ActivationKeysTable/components/Header/index.js

```diff
@@ -33,6 +35,29 @@ const ActivationKeysTableHeader = ({
 }) => {
 	const [activationKeys, setActivationKeys] = activationKeysState;
 
+	const {
+		userAccountsState: [userAccounts],
+	} = useGetAccountUserAccount(project);
+
+	const [isAdminOrPartnerManager, setIsAdminOrPartnerManager] = useState();
+
+	const filteredAccounts = userAccounts?.filter((userAccount) =>
+		userAccount?.roles?.some(
+			(role) =>
+				role === ROLE_TYPES.admin.key ||
+				role === ROLE_TYPES.partnerManager.key
+		)
+	);
+
+	useEffect(() => {
```

Comment:
Esses estados são necessários? Já que já temos uma variável que faz o filtro de ver se tem admin ou partnerManager?

Reply:
O filtro que verifica se é adm ou partnerManager retorna um array vazio ou não. O estado foi para setar um booleano (true or false) a partir desse retorno do array. Se for adm ou partnerManager, retorna true. Caso contrário, retorna false.

Reply:
você usa esse array pra mais alguma coisa? Porque o some retorna um boolean

Reply:
A variável que filtra retorna um array vazio [ ] caso o usuário não seja admin/partnerManager. Caso o usuário seja admin/partnerManager, ele retorna todas as contas que possuam a roles=admin/partnerManager, e não o boolean, que é o que eu preciso para poder setar as condições.